drivers/lxc/driver.go

```diff
@@ -0,0 +1,510 @@
+		exitResult: &drivers.ExitResult{},
+
+		totalCpuStats:  stats.NewCpuStats(),
+		userCpuStats:   stats.NewCpuStats(),
+		systemCpuStats: stats.NewCpuStats(),
+	}
+
+	d.tasks.Set(taskState.TaskConfig.ID, h)
+
+	go h.run()
+	return nil
+}
+
+func (d *Driver) StartTask(cfg *drivers.TaskConfig) (*drivers.TaskHandle, *cstructs.DriverNetwork, error) {
+	if _, ok := d.tasks.Get(cfg.ID); ok {
+		return nil, nil, fmt.Errorf("task with ID %q already started", cfg.ID)
+	}
+
+	var driverConfig TaskConfig
+	if err := cfg.DecodeDriverConfig(&driverConfig); err != nil {
+		return nil, nil, fmt.Errorf("failed to decode driver config: %v", err)
+	}
+
+	d.logger.Info("starting lxc task", "driver_cfg", hclog.Fmt("%+v", driverConfig))
+	handle := drivers.NewTaskHandle(pluginName)
+	handle.Config = cfg
+
+	c, err := d.initializeContainer(cfg, driverConfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	opt := toLXCCreateOptions(driverConfig)
+	if err := c.Create(opt); err != nil {
+		return nil, nil, fmt.Errorf("unable to create container: %v", err)
+	}
+
+	cleanup := func() {
+		if err := c.Destroy(); err != nil {
+			d.logger.Error("failed to clean up from an error in Start", "error", err)
+		}
+	}
+
+	if err := d.configureContainerNetwork(c); err != nil {
+		cleanup()
+		return nil, nil, err
+	}
+
+	if err := d.mountVolumes(c, cfg, driverConfig); err != nil {
+		cleanup()
+		return nil, nil, err
+	}
+
+	if err := c.Start(); err != nil {
+		cleanup()
+		return nil, nil, fmt.Errorf("unable to start container: %v", err)
+	}
+
+	if err := d.setResourceLimits(c, cfg); err != nil {
+		cleanup()
+		return nil, nil, err
+	}
+
+	pid := c.InitPid()
+
+	h := &taskHandle{
+		container:  c,
+		initPid:    pid,
+		taskConfig: cfg,
+		procState:  drivers.TaskStateRunning,
+		startedAt:  time.Now().Round(time.Millisecond),
+		logger:     d.logger,
+
+		totalCpuStats:  stats.NewCpuStats(),
+		userCpuStats:   stats.NewCpuStats(),
+		systemCpuStats: stats.NewCpuStats(),
+	}
+
+	driverState := TaskState{
+		ContainerName: c.Name(),
+		TaskConfig:    cfg,
+		StartedAt:     h.startedAt,
+	}
+
+	if err := handle.SetDriverState(&driverState); err != nil {
+		d.logger.Error("failed to start task, error setting driver state", "error", err)
+		cleanup()
+		return nil, nil, fmt.Errorf("failed to set driver state: %v", err)
+	}
+
+	d.tasks.Set(cfg.ID, h)
+
+	go h.run()
+
+	return handle, nil, nil
+}
+
+func (d *Driver) WaitTask(ctx context.Context, taskID string) (<-chan *drivers.ExitResult, error) {
+	handle, ok := d.tasks.Get(taskID)
+	if !ok {
+		return nil, drivers.ErrTaskNotFound
+	}
+
+	ch := make(chan *drivers.ExitResult)
+	go d.handleWait(ctx, handle, ch)
+
+	return ch, nil
+}
+
+func (d *Driver) handleWait(ctx context.Context, handle *taskHandle, ch chan *drivers.ExitResult) {
+	defer close(ch)
+
+	ticker := time.NewTicker(2 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-d.ctx.Done():
+			return
+		case <-ticker.C:
```

Comment:
This is the strangest way to monitor for the process exiting, but it faithfully reproduces the old driver so :+1: 

Lets at least add a comment so future readers don't think you're responsible for this design. :sweat_smile: 

```go
//XXX Should probably use Process.Wait() instead of polling but this code was ported from pre-plugin drivers
```

Reply:
Whoa! Great find!